openssl-3.2.1 and openssl-3.3.0
--- openssl-3.2.1.old/include/openssl/e_os2.h
+++ openssl-3.2.1.new/include/openssl/e_os2.h
@@ -248,6 +248,8 @@ typedef int int32_t;
 typedef unsigned int uint32_t;
 typedef __int64 int64_t;
 typedef unsigned __int64 uint64_t;
+#define UINT64_C(val) val##ui64
+#define EISCONN         113
 # elif defined(OPENSSL_SYS_TANDEM)
 #  include <stdint.h>
 #  include <sys/types.h>
